# Analysis

This directory contains several markdown files in `.md` format, each of which describes the order of necessary scripts to reproduce figures and tables in the main text and supplement.

## Main Text Figures

* [Figure 1: Nucleotide diversity within and between hosts in rural African and North American gut microbiomes.](./figure_1.md)
* [Figure 2: Schematic representation of putative demographic models.](./figure_2.md)
* [Figure 3. Demographic inference of common commensal gut microbes.](./figure_3.md)
* [Figure 4: Phylogenetically sorted distributions of fitness effects.](./figure_4.md)
* [Figure 5: Differences in selective effects between core and accessory genes.](./figure_5.md)

## Supplemental Figures

* [Figure S1: Akaike information criterion for demographic model selection](./figure_s1.md)
* [Figure S2: Effects of control and downsampling on the site-frequency spectra](./figure_s2.md)
* [Figure S3: SFSs and likelihood surfaces for demographic inference over core genes](./figure_s3.md)
* [Figure S4: Distribution of Î½ and  with 95% confidence intervals over core genes](./figure_s4.md)
* [Figure S5: Phylogenetically sorted neu+gamma DFE over core genes](./figure_s5.md)
* [Figure S6: Difference matrices for LRT of core genes between species](./figure_s6.md)
* [Figure S7: Empirical site-frequency spectra of accessory and core genes](./figure_s7.md)
* [Figure S8: SFSs and likelihood surfaces for demographic inference from accessory genes](./figure_s8.md)
* [Figure S9: Demography of core genes compared to core+accessory genes](./figure_s9.md)
* [Figure S10: Phylogenetically ordered analysis of core and accessory gene DFE](./figure_s10.md)
* [Figure S11: Distribution of N<sub>Curr</sub> for core genes](./figure_s11.md)
* [Figure S12: Number of quasi-phaseable samples in top-level clade of core genes per species](./figure_s12.md)
* [Figure S13: Correlation between mean relative abundance and prevalence vs. N<sub>Curr</sub>](./figure_s13.md)
* [Figure S14: Comparison of N<sub>Anc</sub> between core and accessory genes](./figure_s14.md)

## Supplemental Tables

* [Table S1: Accession numbers for North American data](./table_s1.md)
* [Table S2: Accession numbers for African data](./table_s2.md)
* [Table S3: Demographic model parameters, log likelihood, and Akaike Information Criterion for core genes](./table_s3.md)
* [Table S4: Upper and lower bounds of 95% confidence intervals for  and  under two-epoch demographic model](./table_s4.md)
* [Table S5: DFE model parameters, log likelihood, and Akaike Information Criterion for core genes](./table_s5.md)
* [Table S6: Demographic model parameters, log likelihood, and Akaike Information Criterion for accessory genes](./table_s6.md)
* [Table S7: DFE model parameters, log likelihood, and Akaike Information Criterion for accessory genes](./table_s7.md)
* [Table S8: LRT test results for core vs. accessory genes](./table_s8.md)
